bin/faucet/src/server/pow.rs

```diff
@@ -87,11 +87,7 @@ impl PowParameters {
         let mut challenges =
             challenge_cache.challenges.lock().expect("PoW challenge cache lock poisoned");
 
-        if let Some(challenge) = challenges.get(&self.pow_seed) {
-            if challenge.server_signature != self.server_signature {
-                return Err(InvalidRequest::ServerSignaturesDoNotMatch);
-            }
-        } else {
+        if challenges.get(&self.pow_seed).is_some() {
             return Err(InvalidRequest::InvalidChallenge);
```

Comment:
With this change maybe the variant could be renamed as well to `ChallengeAlreadyUsed` (and the error message changed as wel)